src/lightning/data/cache/dataloader.py

```diff
@@ -154,13 +154,27 @@ def __init__(self, global_rank: int, profile: bool = False) -> None:
         self._global_rank = global_rank
         self._profile = profile
 
-    def __call__(self, dataset_kind: _DatasetKind, *args: Any, **kwargs: Any) -> None:
+    def __call__(
+        self,
+        dataset_kind: Any,
+        dataset: Any,
+        index_queue: Any,
+        data_queue: Any,
+        done_event: Any,
+        auto_collation: Any,
+        collate_fn: Any,
+        drop_last: Any,
+        base_seed: Any,
+        init_fn: Any,
+        worker_id: Any,
+        *args: Any,
+        **kwargs: Any,
```

Comment:
please add proper typing here

Reply:
This is PyTorch internals. Doesn't really matter as it isn't typed there.